### PR TITLE
Project: Weather app

### PR DESCRIPTION
**Pull request by SmithVyne**
## Weather App
  This is my implementation of The Odin Project weather app assignment.

**What does this PR do?**
- This PR is for the TSEs to code review the project

- The project is hosted [here](https://smithvyne.github.io/WorldWeatherApp/)

Follow the steps below to test the project locally. 
First ensure you have a version of [Node.js](http://nodejs.org/) equal or greater than v6.10.0 installed.

1. Clone the repository from a terminal `https://github.com/SmithVyne/WorldWeatherApp.git` and cd into the directory.
3. Install project dependencies by running `npm install`
4. Run `npm run webpack` to compile a development build.
5. Run `npm run start` to serve the project live on `http://localhost:9000` and voila!!!.

### CODE REVIEW

**Review by wathigo — CHANGES_REQUESTED** [review]
### Changes Required :heavy_multiplication_x: 
Great job on this project.
Let's fix some issues before we mark this as done.

**REQUIRED.**
* It has a toggle control to determine if the result will be shown in Celsius or Fahrenheit (There are two valid solutions: students can make two different requests for the Celsius data and the Fahrenheit data OR make only one request and use math to convert it to the other unit) 
Currently, one can only send one request either in celsius or Fahrenheit but the toggle functionality is not working. ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/javascript/weather_app/search_requirement.png) from Odin).
* In `webpack-cli` version `4.0.0`,  the run command `npm run start` does not work. This is an issue addressed [here](https://github.com/webpack/webpack-dev-server/issues/2029). Let's change `"webpack-cli": "^4.0.0"` to `"webpack-cli": "^3.3.11"` in our package.json to solve this issue. Remember to delete `package-lock.json` after updating `package.json` and before running `npm install`.
* It changes its appearance based on the received data ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/javascript/weather_app/appearance_requirement.png) from [Odin](https://www.theodinproject.com/courses/javascript/lessons/weather-app)).

**After making the changes, submit another request for review.**

**Happy Hacking :computer: 
Simon Wathigo.**

**Comment by SmithVyne** [comment]
> ### Changes Required ✖️
> Great job on this project.
> Let's fix some issues before we mark this as done.
> 
> **REQUIRED.**
> 
> * It has a toggle control to determine if the result will be shown in Celsius or Fahrenheit (There are two valid solutions: students can make two different requests for the Celsius data and the Fahrenheit data OR make only one request and use math to convert it to the other unit)
>   Currently, one can only send one request either in celsius or Fahrenheit but the toggle functionality is not working. ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/javascript/weather_app/search_requirement.png) from Odin).
> * In `webpack-cli` version `4.0.0`,  the run command `npm run start` does not work. This is an issue addressed [here](https://github.com/webpack/webpack-dev-server/issues/2029). Let's change `"webpack-cli": "^4.0.0"` to `"webpack-cli": "^3.3.11"` in our package.json to solve this issue. Remember to delete `package-lock.json` after updating `package.json` and before running `npm install`.
> * It changes its appearance based on the received data ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/javascript/weather_app/appearance_requirement.png) from [Odin](https://www.theodinproject.com/courses/javascript/lessons/weather-app)).
> 
> **After making the changes, submit another request for review.**
> 
> **Happy Hacking 💻 Simon Wathigo.**

**Comment by SmithVyne** [comment]
Hello @wathigo, there  is an issue with the instructions you're using. They are outdated. Please use the new updated instructions in [Odin Project ](https://www.theodinproject.com/courses/javascript/lessons/weather-app)

**Review by rOluochKe — CHANGES_REQUESTED** [review]
Hello @SmithVyne 
### You have done well so far :clap: but you have a minor change to make
- [ ] Add a toggle button to show if the result will be shown in Celsius or Fahrenheit as recommended by the previous reviewer. 
### Make the change and submit for another code review. Good luck.
- Happy coding!

File: README.md [hunk 1]
```diff
@@ -1 +1,35 @@
...
+  - Javascript
+  - HTML
+  - CSS
+
+## Installation
+Follow the steps below to test the project locally. First ensure you have a version of [Node.js](http://nodejs.org/) equal or greater than v6.10.0 installed.
+
+1. Clone the repository from a terminal `https://github.com/SmithVyne/WorldWeatherApp.git` and cd into the directory.
+3. Install project dependencies by running `npm install`
+4. Run `npm run webpack` to compile a development build.
```
Comment (christianotieno) [inline README.md:24]: ## Expected change

Isn't this supposed to be `build`? 🤔 

```suggestion
4. Run `npm run build` to compile a development build.
```

**Review by christianotieno — CHANGES_REQUESTED** [review]
### Changes required 🍾 
Hello @SmithVyne 👋 
Good job working on the suggested changes by @wathigo & @rOluochKe 👍 
There is still one small change left to accomplish. 👇 
- [ ] During the calls (API calls) data received should have a default value, either `Celsius` or `Kelvin`. The conversion should **only** happen upon toggling the values getting displayed (`DOM manipulation`) not during fetching. You are expected to write a logic for this conversion so that this functionality runs smoothly. That's what was required in the first two reviews. ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/javascript/weather_app/search_requirement.png) from [Odin](https://www.theodinproject.com/courses/javascript/lessons/weather-app)).

- I hope this helps in clearing things out for you now. 

### Happy coding 💻 

#### Christian Otieno (TSE)

**Comment by SmithVyne** [comment]
Hello, I've been able to implement this now

**Review by rOluochKe — APPROVED** [review]
Hello @SmithVyne 
### You got it right, well done :clap: The project has been approved. Congratulations :boom: :100:
### Merge and submit the project. Good luck with the next one.
- Happy coding!

**Pull request merged**